tree/ntuple/v7/src/RPageStorageFile.cxx

```diff
@@ -220,7 +220,81 @@ ROOT::Experimental::Detail::RPageSourceFile::RPageSourceFile(std::string_view nt
+         [](const RNTupleMetrics &metrics) -> std::pair<bool, double> {
+            if (const auto szReadPayload = metrics.GetCounter("RPageSourceFile.szReadPayload")) {
+               if (const auto szReadOverhead = metrics.GetCounter("RPageSourceFile.szReadOverhead")) {
+                  if (const auto timeWallRead = metrics.GetCounter("RPageSourceFile.timeWallRead")) {
+                     if (auto walltime = timeWallRead->GetValueAsInt()) {
+                        double payload = szReadPayload->GetValueAsInt();
+                        double overhead = szReadOverhead->GetValueAsInt();
+                        // unit: bytes / nanosecond = GB/s; MiBi = GB * 1000 / 1.024 / 1.024
+                        return {true, 1000. * (payload + overhead) / (1.024 * 1.024 * walltime)};
+                     }
+                  }
+               }
+            }
+            return {false, -1.};
+         }
+      ),
+      *fMetrics.MakeCounter<RNTupleCalcFloatPerf*> ("bwReadUnzip", "MiBi/s", "bandwidth uncompressed bytes read per second",
+         [](const RNTupleMetrics &metrics) -> std::pair<bool, double> {
+            if (const auto szReadPayload = metrics.GetCounter("RPageSourceFile.szReadPayload")) {
+               if (const auto szReadOverhead = metrics.GetCounter("RPageSourceFile.szReadOverhead")) {
+                  if (const auto timeWallRead = metrics.GetCounter("RPageSourceFile.timeWallRead")) {
+                     if (auto walltime = timeWallRead->GetValueAsInt()) {
+                        double payload = szReadPayload->GetValueAsInt();
+                        double overhead = szReadOverhead->GetValueAsInt();
+                        // unit: bytes / nanosecond = GB/s; MiBi = GB * 1000 / 1.024 / 1.024
+                        return {true, 1000. * (payload + overhead) / (1.024 * 1.024 * walltime)};
+                     }
+                  }
+               }
+            }
+            return {false, -1.};
+         }
+      ),
+      *fMetrics.MakeCounter<RNTupleCalcFloatPerf*> ("bwUnzip", "MiBi/s", "decompression bandwidth of compressed bytes per second",
+         [](const RNTupleMetrics &metrics) -> std::pair<bool, double> {
+            if (const auto szUnzip = metrics.GetCounter("RPageSourceFile.szUnzip")) {
+                  if (const auto timeWallUnzip = metrics.GetCounter("RPageSourceFile.timeWallUnzip")) {
+                     if (auto walltime = timeWallUnzip->GetValueAsInt()) {
+                        double unzip = szUnzip->GetValueAsInt();
+                        // unit: bytes / nanosecond = GB/s; MiBi = GB * 1000 / 1.024 / 1.024
+                        return {true, 1000. * unzip / (1.024 * 1.024 * walltime)};
+                     }
+                  }
+               }
+            }
+            return {false, -1.};
+         }
+      ),
+      *fMetrics.MakeCounter<RNTupleCalcFloatPerf*> ("rtReadOverhead", "", "ratio of overhead over all bytes read",
```

Comment:
```suggestion
      *fMetrics.MakeCounter<RNTupleCalcFloatPerf*> ("rtReadEfficiency", "", "ratio of payload over all bytes read",
```

Reply:
The measure of course hinges on the definition of what is the "payload", i.e. what's really required to read.  At the moment, we consider necessary all pages that belong to active columns. We could also measure how many pages we read that are eventually never accessed.

Reply:
Not sure what I can do about that - the value "payload" / "overhead" existed already before.